sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -937,6 +944,14 @@ private[execution] object AggregateProcessor {
       imperatives.toArray,
       trackPartitionSize)
   }
+
+  // Rewrites the window partition size attribute in all `SizeBasedWindowFunction`s using executor
+  // side `SizeBasedWindowFunction.n`.
+  def rewriteWindowPartitionSize(expression: Expression): Expression = {
+    expression.transformDown {
+      case f: SizeBasedWindowFunction => f.withPartitionSize(SizeBasedWindowFunction.n)
```

Comment:
Minor: we could also create a BoundReference directly. It is always at position 0.